### DISCUSSION TITLE
Interpolation error

### DISCUSSION THREAD

**Discussion by MiriamSterl** [Parcels-code/Parcels#1238] [Q&A] [answered]
What can cause an InterpolationError during particle advection (with pset.execute)? I am using JITParticles and the source code says that JIT does not yet provide the context that creates such an exception, so I would be interested what possible causes could be, and how to troubleshoot this.

I am using a doubly periodic domain, where the x and y coordinates run from -L/2 to +L/2, so they are not lat/lon coordinates. The InterpolationErrors always occur when a particle reaches one of the lateral boundaries. However, the velocity fields I used for creating the FieldSet are doubly periodic (checked), and I set `fieldset.add_periodic_halo(zonal=True,meridional=True)` when creating the FieldSet.
The first and last columns/rows of the FieldSet are not identical; should they be, or should parcels itself 'glue them together' when given the `add_periodic_halo`?
Or are there any other ideas as to what could be the problem?

**Comment by VeckoTheGecko** [comment, marked as answer]
[This tutorial notebook should help](https://nbviewer.jupyter.org/github/OceanParcels/parcels/blob/master/parcels/examples/tutorial_periodic_boundaries.ipynb). The `.add_periodic_halo()` function only copies a small portion of the fieldset to the other sides of the domain ([documentation](https://oceanparcels.org/gh-pages/html/index.html#parcels.field.Field.add_periodic_halo)), you will need to implement the periodicity directly through the use of custom kernels.

**Reply by MiriamSterl** [reply]
Thank you! I didn't realise the extra kernel was needed, but now it works perfectly :)
